### DISCUSSION TITLE
원시 값 포장을 하는 이유가 무엇일까?

### DISCUSSION THREAD

**Discussion by alstn113** [2024-woowacourse-study/level-interview#34] [OOP]
.

**Comment by gyungchan-jo** [comment]
1. 원시값 포장을 하면, 객체를 사용해야 하는 경우에 늘 내부에서 검증하기 때문에, 비즈니스 로직 전반에서 포장값을 사용함으로서 항상 신뢰를 확보할 수 있어 더 안정적이고 명확한 도메인을 만들 수 있다. 

2. 원시값 포장을 통해 유연하게 객체 내의 타입을 관리할 수도 있다.

**Comment by le2sky** [comment]
원시 값 하나 자체는 그냥 아무 의미 없는 값일 뿐입니다. 
어떤 메서드가 int 값을 매개변수로 받는다면 해당 메서드는 해당 매개변수의 의도를 나타내기 위해 모든 수단과 방법을 가리지 않아야 합니다.

원시 값을 포장하면 다음과 같은 이점을 누릴 수 있습니다.
값이 값 이상의 의미를 가질 수 있다면, 다른 개발자에게 의도를 전달할 수 있습니다. 
해당 값을 왜 사용하지에 대한 정보를 표현할 수 있습니다.
원시타입을 포장함으로써 메서드의 시그니처를 명확하게 표현할 수 있습니다.

**Comment by GIVEN53** [comment]
원시값에 의미 부여
의도 전달
타입관리
